src/App.js

```diff
@@ -227,14 +279,77 @@ export class _AppStackWrapper extends React.Component {
     this.props.updateHeight(height);
   }
 
+  /**
+   * Method called when WebSocketHandler from lib emits a 'wallet' event
+   * We check if there is a new transaction to this wallet
+   *
+   * @param {Object} wsData The message object
+   */
+  handleWebsocketMsg = (wsData) => {
+    if (wsData.type === 'wallet:address_history') {
+      // TODO we also have to update some wallet lib data? Lib should do it by itself
+      const data = hathorLib.wallet.getWalletData();
+      const historyTxs = data.historyTransactions || {};
+      const allTokens = 'allTokens' in data ? data.allTokens : [];
+      hathorLib.wallet.updateHistoryData(historyTxs, allTokens, [wsData.history], null, data);
+
+      const newWalletData = hathorLib.wallet.getWalletData();
+      const { keys } = newWalletData;
+      this.props.newTx(wsData.history, keys);
+    }
+  }
+
+  /**
+   * Method called when WebSocket updates isOnline attribute, so we update this parameter in redux
+   *
+   * @param {boolean} value Boolean if websocket is online
+   */
+  isOnlineUpdated = (value) => {
+    this.props.setIsOnline(value);
+  }
+
+  /**
+   * Update tokens on redux with data from storage, so user doesn't need to add the tokens again
+   */
+  updateReduxTokens = () => {
+    this.props.setTokens(hathorLib.tokens.getTokens());
+  }
+
+  /**
+   * Handles when app changes its state to/from background and foreground.
+   *
+   * More info: https://reactnative.dev/docs/appstate
+   */
+  _handleAppStateChange = (nextAppState) => {
+    if (nextAppState === 'active') {
+      if (this.appState === 'inactive') {
+        // inactive state means the app wasn't in background, so no need to lock
+        // the screen. This happens when user goes to app switch view or maybe is
+        // asked for fingerprint or face if
+        this.backgroundTime = null;
+      } else if (Date.now() - this.backgroundTime > LOCK_TIMEOUT) {
+        // this means app was in background for more than LOCK_TIMEOUT seconds,
+        // so display lock screen
+        this.props.lockScreen();
```

Comment:
If the app gets inactive before starting the wallet, what this `lockScreen=true` will do? Will it redirect to the PinScreen? 

I believe one of the reasons this was in the `Dashboard` is because we should lock the screen only after the user has started the wallet. So we just need to test if this doesn't bring any problems.

We have a variable in the lib of `wallet:loaded`, that could help here.

Reply:
This code is inside `AppStackWrapper`, which in turn is in a switch navigator:
```js
const SwitchNavigator = createSwitchNavigator({
  Decide: DecideStackScreen,
  App: AppStackWrapper,
  Init: InitStack,
}, {
  initialRouteName: 'Decide',
});
```

So it will only be executed after we're past the `InitStack`, when the user creates the wallet. If we reset the wallet, `AppStackWrapper` is removed and the unmount method called, so we remove the listeners.

I thought about placing `AppStackWrapper` on a different file, but it depends on other objects from `App.js` (like `AppStack`), so I couldn't find a good way.

Reply:
I could create `AppStack` on a different file:
```js
const AppStack = createStackNavigator({
  Main: TabNavigator,
  About,
  Security,
  ChangePin,
  ResetWallet,
  PaymentRequestDetail,
  RegisterToken: RegisterTokenStack,
  ChangeToken,
  PinScreen: {
    screen: PinScreen,
    navigationOptions: disableSwipeDown,
  },
  CreateTokenStack,
  TokenDetail,
  UnregisterToken,
}, {
  mode: 'modal',
  headerMode: 'none',
});
```

I'm just not sure if that's best or we should keep all navigation code on `App.js`.

Reply:
Ah, it makes sense. I was just confused with all stacks. But it's ok for me